Sources/ApolloCodegenLib/Frontend/GraphQLType.swift

```diff
@@ -2,24 +2,37 @@ import JavaScriptCore
 
 /// A GraphQL type.
 public indirect enum GraphQLType: Hashable {
-  case named(GraphQLNamedType)
+  case entity(GraphQLCompositeType)
+  case scalar(GraphQLScalarType)
+  case `enum`(GraphQLEnumType)
+  case inputObject(GraphQLInputObjectType)
   case nonNull(GraphQLType)
   case list(GraphQLType)
 
   public var typeReference: String {
     switch self {
-    case let .named(type):
+    case let .entity(type as GraphQLNamedType),
+         let .scalar(type as GraphQLNamedType),
+         let .enum(type as GraphQLNamedType),
+         let .inputObject(type as GraphQLNamedType):
```

Comment:
Ooh fancy, I didn't know you could do this if all the associated objects are the same type